popup.js

```diff
@@ -0,0 +1,157 @@
+
+
+function openNewTab(inputValue) {
+    let url = generateIssueUrl(inputValue)
+    chrome.tabs.create({'url': url});
+}
+
+// Puts latest selection to top to make it default one for page reloads
+function makeDropDownSelectionDefault(){
+    // TODO: make general function that takes param and makes it first array element
+    storedKeys = storedKeys.filter(item => item !== projectKeySelect.value);
+    storedKeys.unshift(projectKeySelect.value);
+    storeProjectKeys(storedKeys)
+}
+
+function generateIssueUrl(inputValue) {    
+    return `https://${companyInput.value}.${jiraBaseUrl.value + projectKeySelect.value}${inputValue ? '-' + issueIdInput.value : ''}`
+}
+
+function useStoredOptionsForDisplayInDOM() {
+    chrome.storage.local.get({
+        // TODO: build solution to create key if not existent, build general function
+        // TODO: remove update-DOM logic from loadKeys from chrome storage
+        projectKeys: [],
+    }, function(items) {
+        //Store retrieved options as the selected values in the DOM
+        if (items.projectKeys == "") {
+            controlModalDisplayStyle("block", keysModal)
+        }
+        storedKeys = items.projectKeys
+        projectKeySelect.innerHTML = "";
+        keyList.innerHTML = "";
+        items.projectKeys.map(key => {
+        projectKeySelect.innerHTML += `<option>${key}</option>`
+        keyList.innerHTML += `<div id="keyListElement" class="keylist-element">${key}<div class="delete-btn" id="delete-${key}">&times;</div></div>`
+        //var tempVar = document.getElementById(`delete-${key}`)
+        //console.log(tempVar)
+        document.addEventListener('click', function(e) {
+            if(e.target && e.target.id == `delete-${key}`) {
+                removeProjectKey(key)            
+            }
+        })
+    })
+    });
+}
+
+function addProjectKey(){
+    storedKeys.push(addProjKeyInput.value.toUpperCase())
+    storeProjectKeys(storedKeys)
+    resetInputValue(addProjKeyInput)
+    useStoredOptionsForDisplayInDOM()
+ }
+
+ function removeProjectKey(projectKey){
+    storedKeys = storedKeys.filter(item => item !== projectKey);
+    storeProjectKeys(storedKeys)
+ }
+
+function resetInputValue(input){
+    input.value = ""
+}
+
+function storeProjectKeys(projectKeys){
+    chrome.storage.local.set({
+        projectKeys: projectKeys
+        })
+    useStoredOptionsForDisplayInDOM()
+}
+
+function controlModalDisplayStyle(display, modalType){    
+    modalType.style.display = `${display}`
+}
+
+function generateJiraUrl(){
+    return (`https://<b style="color: green;">${companyInput.value}</b>.${jiraBaseUrl.value}`);
+}
+
+function updateJiraUrlInUrlCompositionDom(){
+    urlComposition.innerHTML = generateJiraUrl();
+}
+
+function storeJiraUrlElements(companyName, jiraBaseUrl){
+    chrome.storage.local.set({
+        companyName: companyName,
+        jiraBaseUrl: jiraBaseUrl
+        })
+}
+
```

Comment:
I modified the following function from help I got from stackoverflow. I have heard times understanding callbacks/async and guess I'll need to read EloquentJS on it thoroughly again. Do you have any other ressources to consult to understand the concept/train this?

I heard about promisis as well but I think they are not allowed in chrome extensions either.